packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -194,18 +194,19 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
     } as AuthenticateRequestOptions;
     const requestState = await clerkClient.authenticateRequest(req, authenticateRequestOptions);
 
-    if (requestState.status === AuthStatus.Handshake) {
-      const locationHeader = requestState.headers.get('location');
-      if (!locationHeader) {
-        throw new Error('Unexpected handshake without redirect');
-      }
+    const locationHeader = requestState.headers.get('location');
+    if (locationHeader) {
```

Comment:
There are a few scenarios where we pass a location header when the status is not a handshake. Notably, in development we strip the handshake query parameters.